### PR TITLE
[µTVM] Add platform timer and RPCTimeEvaluator to enable AutoTVM

### PR DESCRIPTION
**Pull request by areusch**
This PR adds a platform-specific timing function plus a device-side implementation of RPCTimeEvaluator. In a follow-on, an AutoTVM test will be added (there are a few different dependent PRs in flight here so need to let them all land before I can test that properly).

### CODE REVIEW

**Comment by areusch** [comment]
@tqchen @liangfu @tom-gall @u99127 @manupa-arm @weberlo

File: src/runtime/crt/host/crt_config.h [hunk 1]
```diff
@@ -43,7 +43,7 @@
 #define TVM_CRT_MAX_REGISTERED_MODULES 2
 
 /*! Size of the global function registry, in bytes. */
-#define TVM_CRT_GLOBAL_FUNC_REGISTRY_SIZE_BYTES 200
+#define TVM_CRT_GLOBAL_FUNC_REGISTRY_SIZE_BYTES 256
```
Comment (tmoreau89) [inline src/runtime/crt/host/crt_config.h:46]: out of curiosity, what's the reason for increasing the registry size? just having a nice power of 2?

**Review by tmoreau89 — APPROVED** [review]
Thanks for the PR @areusch, LGTM

**Comment by tmoreau89** [comment]
@liangfu care to take a pass? thanks!

**Comment by liangfu** [comment]
ping @areusch

File: src/runtime/crt/common/crt_runtime_api.c [hunk 2]
```diff
@@ -351,10 +375,120 @@ tvm_crt_error_t TVMInitializeRuntime() {
...
+  TVMByteArray* result_byte_arr;
+  tvm_crt_error_t err =
+      TVMPlatformMemoryAllocate(sizeof(TVMByteArray), result_byte_ctx, (void*)&result_byte_arr);
+  if (err != kTvmErrorNoError) {
+    return err;
+  }
+  size_t data_size = sizeof(double) * g_time_evaluator_state.repeat;
+  err = TVMPlatformMemoryAllocate(data_size, result_byte_ctx, (void*)&result_byte_arr->data);
+  if (err != kTvmErrorNoError) {
+    return err;
```
[the remark this answers is not in the document]
Comment (areusch) [reply src/runtime/crt/common/crt_runtime_api.c]: good suggestion, done.

**Comment by areusch** [comment]
sorry @liangfu for the delay. i've addressed your comment and sync'd to main.

**Review by liangfu — APPROVED** [review]
Thanks for the update, the changes looks good.

**Comment by areusch** [comment]
@tqchen I think this is ready to merge

**Comment by tqchen** [comment]
@liangfu @tmoreau89 feel free to merge given both committer approves :)

**Pull request merged**